### ISSUE TITLE
Expandable Tree view nodes are not clickable with space or enter key and screen reader spoken feedback is not appropriate

### ISSUE THREAD

**Issue by gkhadK** [vmware/clarity#4661]
**Clarity Version:** 3.1.1
**WCAG 2.1 Guideline:** 2.1.1 Keyboard and 4.1.2 Name, role, value.

**Test Environment:**
- Test URL: https://clarity.design/documentation/treeview
- OS: Windows 10
- User Agents/Assistive Technologies: Firefox 76.0
- Screen reader: NVDA 2020.1

**Steps to reproduce:**
1. Open the test URL.
2. Navigate to the different tree node and try to interact, collapse/expand the parent node or any other expandable next level node using space or enter key.

**Expected result:**
Expandable tree nodes should be clickable with space or enter key. While navigating through the tree nodes Name, role and value should be announced appropriately.

**Actual result:**
- With space key the page is scrolled a bit and with enter key nothing happens.
- Collapsed tree node underlying content is getting announced by NVDA
- Indeterminate checkbox Tree level 1 node expand/collapse state is not announced, if the user toggles it more than once.
- Level 2 and level 3 items numbers are announced as 1 of 1 in focus mode. Tab to the parent node (here the earcon for focus mode), then use down arrow keys to navigate through the tree nodes.
- Lazy loading and selection/ recursive trees: Unselected tree nodes are announced as unselected notselected&quot;

**Comment by nghiado** [comment]
Any plan to fix this one soon?
